pkg/scalers/datadog_scaler.go

```diff
@@ -272,29 +290,35 @@ func (s *datadogScaler) getQueryResult(ctx context.Context) (float64, error) {
 
 	series := resp.GetSeries()
 
-	if len(series) > 1 {
-		return 0, fmt.Errorf("query returned more than 1 series; modify the query to return only 1 series")
-	}
-
 	if len(series) == 0 {
 		if !s.metadata.useFiller {
 			return 0, fmt.Errorf("no Datadog metrics returned for the given time window")
 		}
 		return s.metadata.fillValue, nil
 	}
 
-	points := series[0].GetPointlist()
-
-	index := len(points) - 1
-	if len(points) == 0 || len(points[index]) < 2 {
-		if !s.metadata.useFiller {
-			return 0, fmt.Errorf("no Datadog metrics returned for the given time window")
+	// Collect all latest point values from any/all series
+	results := make([]float64, len(series))
+	for i := 0; i < len(series); i++ {
+		points := series[i].GetPointlist()
+		if len(points) == 0 || len(points[0]) < 2 {
+			if !s.metadata.useFiller {
+				return 0, fmt.Errorf("no Datadog metrics returned for the given time window")
+			}
+			return s.metadata.fillValue, nil
 		}
-		return s.metadata.fillValue, nil
+		// Return the last point from the series
+		index := len(points) - 1
+		results[i] = *points[index][1]
 	}
 
-	// Return the last point from the series
-	return *points[index][1], nil
+	switch s.metadata.queryAggegrator {
+	case avgString:
+		return AvgFloatFromSlice(results), nil
+	default:
```

Comment:
I wouldn't do a default. If the user hasn't added the queryAggregator field, then they don't really know what they are doing, so I would return an error here: "query returned more than 1 series; modify the query to return only 1 series or add a queryAggregator"

Reply:
It only applies in cases where a user has defined a query containing multiple queries, separated by comma -- I think they'd expect something to happen if they do that? I expect the Warn log above would be sufficient perhaps.

More importantly though, for code simplicity, this code runs even if only a single series is returned, and ```MaxFloatFromSlice()``` will just return the first/single value - it saves some code/logic for single-vs-multi results this way. Not having a default means we'd have different code paths depending on the number of returned ```Series``` elements.

Reply:
Fixed this behaviour

Reply:
👍